osu.Server.Spectator/Hubs/CountdownImplementation.cs

```diff
@@ -102,6 +110,20 @@ async Task start()
             }
         }
 
+        public async Task UpdateTimeRemaining()
+        {
+            using (var roomUsage = await hub.GetRoom(roomId))
```

Comment:
I feel really weird about this reverse retrieval lock lookup thing.

Rather than having this `GetRoom` logic here (which feels pretty weird), how do you feel about making a method in `ServerMultiplayerRoom` like `UpdateTimings` which can perform this?

I almost feel like this whole class is only adding complexity and may be better to be integrated as part of `ServerMultiplayerRoom` itself..

Reply:
Sure. I've merged `CountdownImplementation` into `ServerMultiplayerRoom` and changed it to `UpdateForRetrieval()` for if we want to use this in the future.

I originally didn't do it this way because:
1. I had more complex logic around the countdown `Task`.
2. The async method needs to re-retrieve the room, it can't access the local instance at all. This is a little less obvious now but I've commented it in the places where it's done so that hopefully it won't be missed if this is iterated on in the future. The other alternative is to make the method static but you'd have to pass in 8 parameters directly or via a struct.